server/message_hooks.go

```diff
@@ -454,13 +449,17 @@ func (p *Plugin) SendChat(srcUser string, usersIDs []string, post *model.Post) (
 
 	_, err := p.store.MattermostToTeamsUserID(srcUser)
 	if err != nil {
-		p.handlePromptForConnection(srcUser, post.ChannelId)
+		if chatMembersSpanPlatforms {
+			p.handlePromptForConnection(srcUser, post.ChannelId)
+		}
```

Comment:
We use `chatMembersSpanPlatforms` to decide whether to notify: if it doesn't span, the user is expected to receive it in Mattermost and there's no "need" to connect. (We still try to `SendChat` when `selectiveSync` is disabled if bidirectional sync is expected.)